devolutions-session/src/dvc/process.rs

```diff
@@ -108,11 +107,107 @@ pub struct WinApiProcessCtx {
 }
 
 impl WinApiProcessCtx {
-    // Returns process exit code.
-    pub fn start_io_loop(&mut self) -> Result<u32, ExecError> {
+    pub fn process_abort(&mut self, exit_code: u32) -> Result<(), ExecError> {
+        info!(
+            session_id = self.session_id,
+            "Aborting process execution by user request"
+        );
+
+        self.process.terminate(exit_code)?;
+
+        Ok(())
+    }
+
+    pub fn process_cancel(&mut self) -> Result<(), ExecError> {
+        info!(
+            session_id = self.session_id,
+            "Cancelling process execution by user request"
+        );
+
+        post_message_for_pid(self.pid, WM_QUIT, WPARAM(0), LPARAM(0))?;
+
+        // TODO: Figure out how to correctly send CTRL+C to console applications.
```

Comment:
note: Maybe open a Jira ticket for that?

You can then refer to it like that: `TODO(DGW-XXX):`.